auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/GrpcUmsClient.java

```diff
@@ -379,24 +374,53 @@ public Account getAccountDetails(String actorCrn, String accountId, Optional<Str
         return client.listAssigmentsOfUser(RequestIdUtil.getOrGenerate(requestId), userCrn);
     }
 
-    //"#value.concat(#fieldId).concat(#projectId)"
-//    @Cacheable(cacheNames = "umsUserRightsCache", key = "{ #actorCrn, #userCrn, #right, #resource }")
-    @Cacheable(cacheNames = "umsUserRightsCache", key = "{ #userCrn, #right, #resource }")
-    public boolean checkRight(String actorCrn, String userCrn, String right, String resource, Optional<String> requestId) {
+    @Cacheable(cacheNames = "umsUserHasRightsForResourceCache", key = "{ #actorCrn, #userCrn, #right, #resource }")
+    public boolean checkResourceRight(String actorCrn, String userCrn, String right, String resource, Optional<String> requestId) {
         if (InternalCrnBuilder.isInternalCrn(userCrn)) {
             LOGGER.info("InternalCrn, allow right {} for user {}!", right, userCrn);
             return true;
         }
-        if (!isEntitledAndLogResult(actorCrn, ThreadBasedUserCrnProvider.getAccountId(), Entitlement.CB_AUTHZ_POWER_USERS)) {
-            if (RightUtil.isReadRight(right)) {
-                LOGGER.info("In account {} authorization related entitlement disabled, thus skipping permission check!!",
-                        ThreadBasedUserCrnProvider.getAccountId());
-                return true;
-            } else {
-                // if legacy authz then we will check permission on account level
-                resource = null;
-            }
+        return makeCheckRightCall(actorCrn, userCrn, right, resource, requestId);
+    }
+
+    @Cacheable(cacheNames = "umsUserHasRightsForResourceCache", key = "{ ##actorCrn, userCrn, #right, #resource }")
```

Comment:
i haven't noticed it, but int test brought out this double ##
```
"statusReason":"Error message: \"Internal server error: Expression [{ ##actorCrn, userCrn, #right, #resource }] @3: EL1043E: Unexpected token. Expected 'identifier' but was 'hash(#)'\""
```

Reply:
Thank you!